### PR TITLE
Add the kotlin language's stdlib

### PR DESCRIPTION
**Pull request by Techcable**
Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/Jetbrains/kotlin
 * Official website (optional, not the repository): https://kotlinlang.org/
* NPM package url (optional):https://www.npmjs.com/package/kotlin
 * License and its reference: [Apache 2.0 License](https://github.com/JetBrains/kotlin/blob/master/license/LICENSE.txt)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 649
   - Count of stars: 11380
   - Count of forks: 1104
 * NPM download stats (optional):
   - Downloads in the last day: 11
   - Downloads in the last week: 285
   - Downloads in the last month: 590

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

NOTE: Although kotlin does have a npm package, I didn't include an autoupdate configuration since it's out of date and the library isn't minified. Since the language uses a lot of metadata, I'm not exactly sure if minifying it is safe and it may cause bugs. Even though it's a very popular language, they don't seem to pay much attention to their NPM package and it's a few months out of date.

### CODE REVIEW

File: ajax/libs/kotlin/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "kotlin",
+  "version": "1.1.2-4",
+  "description": "Standard Library for Kotlin Applications",
+  "main": "kotlin.js",
```
Comment (pvnr0082t) [inline ajax/libs/kotlin/package.json]: @Techcable Please remove `main`, `bugs` fields. Thanks.

File: ajax/libs/kotlin/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "Runtime",
+    "JavaScript",
+    "JetBrains"
+  ],
+  "author": "JetBrains",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "kotl.in/issue"
+  },
+  "homepage": "kotlinlang.org"
```
Comment (pvnr0082t) [inline ajax/libs/kotlin/package.json]: Please use `https://kotlinlang.org/` instead. Thanks.

File: ajax/libs/kotlin/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "kotlin",
+  "version": "1.1.2-4",
```
Comment (pvnr0082t) [inline ajax/libs/kotlin/package.json]: Please add the latest stable version. I don't see `v1.1.2-4` published on both GitHub and NPM.

**Comment by Techcable** [comment]
Jetbrains releases the latest version of kotlin javascript [in a jar file on maven central](https://repo1.maven.org/maven2/org/jetbrains/kotlin/kotlin-stdlib-js/). That's where I downloaded the files from, using the update script that I put in the package directory.

**Comment by Techcable** [comment]
I removed the update script like you requested, and applied further fixes to package.json.
However, I have [uploaded the script to a gist](https://gist.github.com/Techcable/22fe59fc2b6755c3d94857d581bac8b7), so you can use it in the future if you want.

**Comment by maruilian11** [comment]
@Techcable Is maven central link official? Could you provide the reference? Thanks.
@pvnr0082t Looks like we need to compile the lib to javascript by our own. 
https://kotlinlang.org/docs/tutorials/javascript/kotlin-to-javascript/kotlin-to-javascript.html
Or we may need to manually add old versions from the maven central site. 
Do you have any idea?

**Comment by Techcable** [comment]
The maven central version is certanially official, [as it's used in many examples](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/resources/testProject/kotlin2JsProject/libraryProject/build.gradle#L17). While you could try and compile from source every update, it'd be much easier to just download and minify from maven central like I did in my script.

**Comment by pvnr0082t** [comment]
@Techcable We usually don't update the library by compiling source ourselves. I think it will be better if the author can publish compiled files on NPM. Then, we can set auto-update config to update library automatically in the future. Could you please discuss with the author? Thanks.

**Comment by sufuf3** [comment]
@Techcable Would you like to finish this PR?
I think the files can add from npm, so you don't have to build files by yourself.

If there is no response in 7 days, I will add this lib ASAP.

Thank you.

**Comment by Techcable** [comment]
I've somewhat lost interest in this, so it'd be nice if you could finish it for me ;)

**Comment by sufuf3** [comment]
@Techcable OK, it's fine.
@pvnr0082t I found this PR is for #11336 and this issue is assigned to you.
So I'll close this PR and let you add this lib(#11336).
Thank you.
cc @PeterDaveHello

**Pull request closed without merging**